src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/ScriptsGitRepoManage.java

```diff
@@ -60,17 +65,31 @@ public void checkoutScripts(String workspace, TerraformScriptGitRepoDetails scri
         } else {
             log.info("Scripts repo is already cloned in the workspace.");
         }
+        return folderContainsScripts(workspace, scriptsRepo);
     }
 
-
-    /**
-     * Recover method for checkoutScripts.
-     *
-     * @param e GitRepoCloneException
-     */
-    @Recover
-    public void recoverCheckoutScripts(GitRepoCloneException e) {
-        log.error("Retry exhausted. Throwing exception: " + e.getMessage());
-        throw e;
+    private List<File> folderContainsScripts(String workspace,
```

Comment:
its better we split this into two ->
getSourceFiles  and then pass it to validateIfFolderContainsTerraformScripts which returns void or throws exception. 
Then simply return the files returned by getSourceFiles.